src/librustc_mir/util/liveness.rs

```diff
@@ -103,18 +103,18 @@ pub struct LivenessMode {
 }
 
 /// A combination of liveness results, used in NLL.
-pub struct LivenessResults<V: LiveVariableMap> {
+pub struct LivenessResults<V> {
     /// Liveness results where a regular use makes a variable X live,
     /// but not a drop.
-    pub regular: LivenessResult<V::LiveVar>,
+    pub regular: LivenessResult<V>,
 
     /// Liveness results where a drop makes a variable X live,
     /// but not a regular use.
-    pub drop: LivenessResult<V::LiveVar>,
+    pub drop: LivenessResult<V>,
 }
 
-impl<V: LiveVariableMap> LivenessResults<V> {
-    pub fn compute<'tcx>(mir: &Mir<'tcx>, map: &dyn LiveVariableMap<LiveVar = V>) -> LivenessResults<V::LiveVar> {
+impl<V, M: LiveVariableMap<LiveVar = V>> LivenessResults<V> {
+    pub fn compute<'tcx>(mir: &Mir<'tcx>, map: &M) -> LivenessResults<V> {
```

Comment:
I would probably make this `map: &impl LiveVariableMap<LiveVar = V>` and remove the explicit `M` parameter altogether